libbeat/publisher/pipeline/output_test.go

```diff
@@ -85,41 +85,51 @@ func TestMakeClientWorker(t *testing.T) {
 	}
 }
 
-func TestMakeClientWorkerAndClose(t *testing.T) {
+func TestReplaceClientWorker(t *testing.T) {
 	tests := map[string]func(mockPublishFn) outputs.Client{
 		"client":         newMockClient,
 		"network_client": newMockNetworkClient,
 	}
 
 	const minEventsInBatch = 50
+	const maxEventsInBatch = 150
 
 	for name, ctor := range tests {
 		t.Run(name, func(t *testing.T) {
 			seedPRNG(t)
 
 			err := quick.Check(func(i uint) bool {
-				numBatches := 1000 + (i % 100) // between 1000 and 1099
+				numBatches := 10000 + (i % 100) // between 1000 and 1099
 
 				wqu := makeWorkQueue()
-				numEvents := atomic.MakeUint(0)
+				retryer := newRetryer(logp.NewLogger("test"), nilObserver, wqu, nil)
+				defer retryer.close()
+
+				var batches []publisher.Batch
+				var numEvents int
+				for i := uint(0); i < numBatches; i++ {
+					batch := randomBatch(minEventsInBatch, maxEventsInBatch).withRetryer(retryer)
+					numEvents += batch.Len()
+					batches = append(batches, batch)
+				}
 
 				var wg sync.WaitGroup
 				wg.Add(1)
 				go func() {
 					defer wg.Done()
-					for batchIdx := uint(0); batchIdx <= numBatches; batchIdx++ {
-						batch := randomBatch(minEventsInBatch, 150, wqu)
-						numEvents.Add(uint(len(batch.Events())))
+					for _, batch := range batches {
 						wqu <- batch
 					}
 				}()
 
 				// Publish at least 1 batch worth of events but no more than 20% events
-				publishLimit := uint(math.Max(minEventsInBatch, float64(numEvents.Load())*0.2))
+				publishLimit := uint(math.Max(minEventsInBatch, float64(numEvents)*0.2))
 
 				var publishedFirst atomic.Uint
 				blockCtrl := make(chan struct{})
+				var batchCount int
```

Comment:
might be a debugging variable that can be removed.

Reply:
Removed in 15a1383.